### ISSUE TITLE
Bug: `derive_vars_dt()` not imputing dates when arguments to both min_dates and max_dates are provided.

### ISSUE THREAD

**Issue by DeclanHodges** [pharmaverse/admiral#1843]
### What happened?

The following call results in dates not being imputed, which appears to be as a result of specifying both min_dates and max_dates in the same call. The same happens when imputing to "first" relative to min_dates, but also specifying max_dates. However, no error/warning is given:
derive_vars_dt(
  dataset = ex,
  dtc = EXENDTC,
  new_vars_prefix = "EXEN",
  highest_imputation = "Y",
  min_dates = exprs(EXSTDTC),
  max_dates = exprs(EOSDT),
  date_imputation = "last",
  flag_imputation = "none"
)

Resulting dataset:
![image](https://user-images.githubusercontent.com/113363343/228888946-4f9a4e2c-c1eb-4854-83df-66ddfd0c098d.png)

### Reproducible Example

ex <- tribble(
  ~USUBJID, ~EXTRT, ~EXDOSE, ~EXSTDTC, ~EXENDTC, ~EOSDT,
  "01-704-1233", "PLACEBO", 0, "2013-03-21", "2013-04-04", "2013-07-14",
  "01-704-1233", "PLACEBO", 0, "2013-04-05", NA, "2013-07-14",
  "01-705-1018", "PLACEBO", 0, "2013-07-05", NA, "2013-07-12",
  "01-705-1031", "XANOMELINE", 54, "2013-11-27", "2013-12-18", "2014-05-11",
  "01-705-1031", "XANOMELINE", 54, "2013-12-19", NA, "2014-05-11"
) %>% 
  mutate(EOSDT = as.Date(EOSDT), 
         EXSTDTC = as.Date(EXSTDTC))

# Non imputing (with min_dates)
example1 <- derive_vars_dt(
  dataset = ex,
  dtc = EXENDTC,
  new_vars_prefix = "EXEN",
  highest_imputation = "Y",
  min_dates = exprs(EXSTDTC),
  max_dates = exprs(EOSDT),
  date_imputation = "last",
  flag_imputation = "none"
)

# Imputing (without min_dates)
example2 <- derive_vars_dt(
  dataset = ex,
  dtc = EXENDTC,
  new_vars_prefix = "EXEN",
  highest_imputation = "Y",
  max_dates = exprs(EOSDT),
  date_imputation = "first",
  flag_imputation = "none"
)

**Comment by zdz2101** [comment]
Hey @DeclanHodges are you sure you're using the newest version of admiral 0.10.1? In your `example2` there actually should be a warning, due to the correspondence of `max_dates` and `date_imputation` = "first" when `highest_imputation` is set to "Y", but I am able to recreate the bug in `example1`

**Comment by DeclanHodges** [comment]
Hey @zdz2101, yes sorry about that, example 2 should say `date_imputation` = "last" (I'll edit). I also do get the warning you mentioned when using `max_dates` and `date_imputation` = "first". I've doubled checked and I am using `admiral` 0.10.1.

**Comment by DeclanHodges** [comment]
> With that said, are you trying to replicate the result of say this:
> 
> example3 <- derive_vars_dt( dataset = ex, dtc = EXENDTC, new_vars_prefix = "EXEN", highest_imputation = "Y", max_dates = exprs(EOSDT), date_imputation = "last", flag_imputation = "none" )
> 
Yes, this is the code that I'm now using (and meant to put as example2!) and it does work to impute the dates as expected. What I wanted to highlight here is that example 1 doesn't impute dates as expected, but also doesn't throw an error/warning, so it took a while to locate the issue in my code. So, I'm suggesting that it could be useful to add an error/warning message if the user attempts to add both `min_dates` and `max_dates` in this scenario.
